### PR TITLE
Add test cases for mathjax commands

### PR DESCRIPTION
**Pull request by physikerwelt**
This adds testcases for all mediwiki specific mathjax
command for client-side mathjax rendering.

### CODE REVIEW

File: test/mathjax-texvc.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+describe('Run test for all mathjax-texvc commands:', function () {
+    this.timeout(0);
+    // read test cases
+    var formulae = require('./mathjax-texvc.json');
+    // create a mocha test case for each chunk
+    formulae.forEach(function (testcase) {
+        if(testcase.ignore !== true ) {
+            it(testcase.id+"$"+testcase.input+"$", function (done) {
```
Comment (d00rman) [inline test/mathjax-texvc.js]: Can do without `done` as well - it's all synchronous

File: test/mathjax-texvc.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                var result = texvcjs.check(testcase.input);
+                assert.equal(result.output, testcase.texvcjs, testcase.id + " failed\n got '" +
```
Comment (d00rman) [inline test/mathjax-texvc.js]: Maybe create a nice JSON and pretty-stringify it? Something like:

```js
JSON.stringify({
  id: testcase.id,
  output: result.output,
  expected: testcase.texvcjs
}, null, 2);
```

File: test/mathjax-texvc.json [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+	},
+	{
+		"id": 87,
+		"input": "\\bold{x}",
+		"texvcjs": "{\\mathbf {x}}"
+	},
+	{
+		"id": 88,
+		"input": "\\href",
+		"ignore": true
```
Comment (d00rman) [inline test/mathjax-texvc.json]: Maybe put a comment stating why is it listed as a test case but ignored.

File: test/mathjax-texvc.json [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+	},
+	{
+		"id": 89,
+		"input": "\\style",
+		"ignore": true
```
Comment (d00rman) [inline test/mathjax-texvc.json]: Idem

**Comment by cscott** [comment]
Merged in https://github.com/cscott/texvcjs/commit/af16957044d344a7571dd7ef0c3fd9721fd4c1c1

**Comment by cscott** [comment]
Is `\hAar` a typo in the upstream texvccheck?  Is there any evidence that this command is used, and isn't just a typo for `hArr` ?

**Comment by cscott** [comment]
(Note also that this patch was mostly unnecessary, as all of these functions are covered explicitly by test cases in `test/all.js`.)

**Comment by physikerwelt** [comment]
@cscott this list was generated from the mathjax extension texvc that was developed over the years, by various people https://github.com/wikimedia/mediawiki-extensions-Math/blob/4d1b9d2b8541134c732d78c9126b83381b9d45ed/modules/mediawiki-extensions/texvc.js that most of it was already converd is a good signal and proves that everything that worked with clientside mathjax before is also accepted by texvcjs (ecept for the ignored commands)

**Pull request merged**